vpr/src/draw/draw.cpp

```diff
@@ -3997,6 +4066,577 @@ void net_max_fanout(GtkWidget* /*widget*/, gint /*response_id*/, gpointer /*data
+
+    //input validity
+    int temps = atoi(gtk_entry_get_text((GtkEntry*)entry));
+    if (temps >= 1 && strchr(gtk_entry_get_text((GtkEntry*)entry), '.') == NULL) {
+        draw_state->list_of_breakpoints.push_back(breakpoint("bt_temps", temps));
+        std::string bpDescription = "Breakpoint at temp_count += " + std::to_string(temps);
+        add_to_bpList(bpDescription);
+    } else
+        invalid_entry_window("Invalid temperature");
+}
+
+//sets a new block type breakpoint
+void set_block_button_callback(GtkWidget* /*widget*/, GtkWidget* grid) {
+    t_draw_state* draw_state = get_draw_state_vars();
+    GtkWidget* entry = gtk_grid_get_child_at((GtkGrid*)grid, 1, 3);
+
+    draw_state->list_of_breakpoints.push_back(breakpoint("bt_from_block", atoi(gtk_entry_get_text((GtkEntry*)entry))));
+    std::string s(gtk_entry_get_text((GtkEntry*)entry));
+    std::string bpDescription = "Breakpoint from_block == " + s;
+    add_to_bpList(bpDescription);
+}
+
+//sets a new router_iter type breakpoint
+void set_router_iter_button_callback(GtkWidget* /*widget*/, GtkWidget* grid) {
+    t_draw_state* draw_state = get_draw_state_vars();
+    GtkWidget* entry = gtk_grid_get_child_at(GTK_GRID(grid), 1, 5);
+
+    //check for input validity
+    int iters = atoi(gtk_entry_get_text((GtkEntry*)entry));
+    if (iters >= 1 && strchr(gtk_entry_get_text((GtkEntry*)entry), '.') == NULL) {
+        draw_state->list_of_breakpoints.push_back(breakpoint("bt_router_iter", iters));
+        std::string bpDescription = "Breakpoint at router_iter == " + std::to_string(iters);
+        add_to_bpList(bpDescription);
+    } else
+        invalid_entry_window("Invalid Router Iteration");
+}
+
+//sets a new net_id type breakpoint
+void set_net_id_button_callback(GtkWidget* /*widget*/, GtkWidget* grid) {
+    t_draw_state* draw_state = get_draw_state_vars();
+    GtkWidget* entry = gtk_grid_get_child_at((GtkGrid*)grid, 1, 6);
+
+    draw_state->list_of_breakpoints.push_back(breakpoint("bt_rouet_net_id", atoi(gtk_entry_get_text((GtkEntry*)entry))));
+    std::string s(gtk_entry_get_text((GtkEntry*)entry));
+    std::string bpDescription = "Breakpoint route_net_id == " + s;
+    add_to_bpList(bpDescription);
+}
+
+//sets a new expression type breakpoint
+void set_expression_button_callback(GtkWidget* /*widget*/, GtkWidget* grid) {
+    t_draw_state* draw_state = get_draw_state_vars();
+    GtkWidget* entry = gtk_grid_get_child_at((GtkGrid*)grid, 1, 2);
+
+    //check input validity
+    std::string expr = gtk_entry_get_text((GtkEntry*)entry);
+    if (valid_expression(expr)) {
+        draw_state->list_of_breakpoints.push_back(breakpoint("bt_expression", expr));
+        std::string bpDescription = "Breakpoint at " + expr;
+        add_to_bpList(bpDescription);
+    } else
+        invalid_entry_window("Invalid expression");
+}
+
+//window that pops up when an entry is not valid
+void invalid_entry_window(std::string error) {
```

Comment:
Might want to add _bp_ somewhere in this and similar function names (best would probably be a bp_ prefix) to make them more distinct.  We may have multiple invalid entries types eventually in the gui.

Reply:
resolved